### PR TITLE
Several small cleanups

### PR DESCRIPTION
**Pull request by dtardon**

### CODE REVIEW

File: src/busctl/busctl-introspect.c [hunk 1]
```diff
@@ -692,36 +692,24 @@ int parse_xml_introspect(const char *prefix, const char *xml, const XMLIntrospec
                 _cleanup_free_ char *name = NULL;
 
                 r = xml_tokenize(&context.current, &name, &context.xml_state, NULL);
-                if (r < 0) {
-                        log_error("XML parse error");
-                        goto finish;
-                }
+                if (r < 0)
+                        log_error_errno(r, "XML parse error");
```
Comment (yuwata) [inline src/busctl/busctl-introspect.c]: Missing `return` ??

File: src/busctl/busctl-introspect.c [hunk 2]
```diff
@@ -692,36 +692,24 @@ int parse_xml_introspect(const char *prefix, const char *xml, const XMLIntrospec
...
-                        log_error("Unexpected token.");
-                        r = -EBADMSG;
-                        goto finish;
-                }
+                                        return r;
+                        } else
+                                log_error_errno(SYNTHETIC_ERRNO(EBADMSG),
+                                                "Unexpected tag '%s' in introspection data.", name);
+                } else if (r != XML_TEXT || !in_charset(name, WHITESPACE))
+                        log_error_errno(SYNTHETIC_ERRNO(EBADMSG), "Unexpected token.");
```
Comment (yuwata) [inline src/busctl/busctl-introspect.c]: Also here. two `return` seem to be missing.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by yuwata** [comment]
BTW, I'd like to cherry-pick the commit for localed-util into #25718, otherwise, the PR conflicts with this PR.

File: src/busctl/busctl-introspect.c (hunk 1 above)
Comment (dtardon) [reply src/busctl/busctl-introspect.c]: Oops, fixed.

File: src/busctl/busctl-introspect.c (hunk 2 above)
Comment (dtardon) [reply src/busctl/busctl-introspect.c]: Yeah...

**Review by yuwata — APPROVED** [review]

**Pull request merged**